File: Javascript/traversy-js-challenges-main/02-basic-challenges-2/08-dice-game/dice-game.js
function diceGameSimulation(numberOfSimulations) {
    gameResults = []
    for (let i = 0; i < numberOfSimulations; i++) {
        const dice1 = Math.floor(Math.random() * 6) + 1;
        const dice2 = Math.floor(Math.random() * 6) + 1;
        const sum = dice1 + dice2;
        const gameResult = {
            "dice1": dice1,
            "dice2": dice2,
            "sum": sum,
            "result": sum === 7 || sum === 11 ? 'win' :
            sum === 2 || sum === 3 || sum === 12 ? 'lose' :
            'roll again'
        }
        gameResults.push(gameResult)
    }
    return gameResults;
}

module.exports = diceGameSimulation;

File: Python/AlgorithmPractice/dice_game/dice_game.py
from random import randrange

class DiceGame:

    def __init__(self, number_of_rolls):
        self.number_of_rolls = number_of_rolls

    def getResults(self):
        results = []
        for x in range(self.number_of_rolls):
            dice_1 = randrange(1, 7)
            dice_2 = randrange(1, 7)
            sum_of_dice = dice_1 + dice_2
            result = {
                       "dice_1": dice_1,
                       "dice_2": dice_2,
                       "sum": sum_of_dice,
                      } 
            if sum_of_dice == 7 or sum_of_dice ==11:
                result["result"] = "win"
            elif sum_of_dice == 2 or sum_of_dice == 3 or sum_of_dice == 12:
                result["result"] = "lose"
            else:
                result["result"] = "roll again"
            results.append(result)
        return results


